src/osp/Active/SysWire.h

```diff
@@ -116,261 +127,488 @@ namespace wiretype
 {
-    if (list() == nullptr)
-    {
-        // Not connected to any WireOutput!
-        return nullptr;
-    }
-    else
+    auto view = rScene.get_registry()
+            .view<osp::active::ACompVehicle,
+                  osp::active::ACompVehicleInConstruction>();
+
+    wire_id_t const id = wiretype_id<WIRETYPE_T>();
+
+    for (auto [vehEnt, rVeh, rVehConstr] : view.each())
     {
-        // Attempt to get value of variant
-        return std::get_if<T>(&(list()->value()));
+        SysWire::construct_panels<WIRETYPE_T>(
+                    rScene, vehEnt, rVeh, *(rVehConstr.m_blueprint));
+
+        signal_construct_nodes(rScene, vehEnt, rVeh, *(rVehConstr.m_blueprint));
     }
 }
 
-inline WireInput::WireInput(IWireElement* element, std::string name) noexcept
- : m_element(element)
- , m_name(std::move(name))
-{ }
+template<typename WIRETYPE_T>
+void SysSignal<WIRETYPE_T>::signal_update_nodes(ActiveScene& rScene)
+{
+    auto &rNodes = rScene.reg_get< ACompWireNodes<WIRETYPE_T> >(rScene.hier_get_root());
+    auto &rWire = rScene.reg_get<ACompWire>(rScene.hier_get_root());
 
-inline WireInput::WireInput(IWireElement *element, WireInput&& move) noexcept
- : LinkedListItem<WireInput, WireOutput>(std::move(move))
- , m_element(element)
- , m_name(std::move(move.m_name))
-{ }
+    // Machines to update accumulated throughout updating nodes
+    std::vector<std::vector<ActiveEnt>> machToUpdate(rWire.m_entToCalculate.size());
 
-//-----------------------------------------------------------------------------
+    // Loop through all nodes that need to be updated
+    for (nodeindex_t<WIRETYPE_T> nodeIndex : rNodes.m_needUpdate)
+    {
+        WireNode<WIRETYPE_T> &rNode = rNodes.get_node(nodeIndex);
+
+        // Overwrite value with value to write
+        rNode.m_state.m_value = rNode.m_state.m_write;
+
+        // Add connected machine inputs to machToUpdate vectors, as they now
+        // require a calculation update
+        for (auto it = std::next(std::begin(rNode.m_links));
+             it != std::end(rNode.m_links); it++)
+        {
+            ActiveEnt ent = it->m_entity;
+            auto const& type = rScene.reg_get<ACompMachineType>(ent);
+            machToUpdate[size_t(type.m_type)].push_back(ent);
+        }
+    }
 
-inline WireOutput::WireOutput(IWireElement* element, std::string name)
- : m_element(element)
- , m_name(std::move(name))
-{ }
-
-inline WireOutput::WireOutput(IWireElement* element, std::string name, WireInput& propagateDepend)
- : m_element(element)
- , m_name(std::move(name))
-{ }
-
-inline WireOutput::WireOutput(IWireElement *element, WireOutput&& move)
- : LinkedList<WireInput>(std::move(move))
- , m_value(std::move(move.m_value))
- , m_element(element)
- , m_name(std::move(move.m_name))
-{ }
+    rNodes.m_needUpdate.clear();
+
+    // Commit machines to update to the scene-wide vectors
+    for (int i = 0; i < machToUpdate.size(); i ++)
```

Comment:
for (size_t const i : std::views::iota(1u, machToUpdate.size()))

Reply:
Or using the ranges-v3 library (most of the rest of which is pending inclusion into the stdlib for 23)
```cpp
for(auto && [rVec, mutex, entToCalculate] : ranges::zip(machToUpdate, rWire.m_entToCalculateMutex, m_entToCalculate))
{
    if( ! rVec.empty())
    {
        std::lock_guard<std::mutex> guard(mutex);
        vecset_merge(entToCalculate, rVec);
        rWire.request_update();
    }
}
```

Reply:
I can almost hear a C programmer laughing at us right now with all these almost-pythonic ways of doing for loops. The cmake is still configured to C++17, and I don't intend to switch the project to C++20 in this PR however, and I'll just keep these in mind.